datasets/sec_failure_to_deliver/pipelines/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,236 @@
+    cmd_list = [
+        # resolve newlines in quoted text
+        f"sed -zi 's/\\n\\\"\\n/\\\"\\n/g' {file}",
+        f"sed -zi 's/\\n\\\",,,,,\\n/\\\",,,,,\\n/g' {file}",
+        f"sed -i '/^\\\",,,,,/d'  {file}",
+        # remove NUL characters
+        "sed -i 's/\\x0//g' " + file,
+        # remove trailer text from all source files under the source path recursively
+        f'find {path} -type f -name "*.csv" -exec sed -i "/Trailer record count/d" {{}} +',
+        f'find {path} -type f -name "*.csv" -exec sed -i "/Trailer total quantity of shares/d" {{}} +',
+    ]
+    logging.info(f"Resolving source data issues on file {file}")
+    for cmd in cmd_list:
+        logging.info("cmd: " + cmd)
+        subprocess.check_call([cmd], shell=True)
+
+
+def process_data(
+    source_file: str,
+    target_file: str,
+    chunksize: str,
+    input_headers: typing.List[str],
+    data_dtypes: dict,
+    output_headers: typing.List[str],
+) -> None:
+    logging.info(f"Processing {source_file} started")
+    with pd.read_csv(
+        source_file,
+        engine="python",
+        encoding="utf-8",
+        quotechar='"',
+        chunksize=int(chunksize),
+        sep=",",
+        names=input_headers,
+        skiprows=1,
+        dtype=data_dtypes,
+    ) as reader:
+        for chunk_number, chunk in enumerate(reader):
+            logging.info(
+                f"Processing chunk #{chunk_number} of file {source_file} started"
+            )
+            target_file_batch = str(target_file).replace(".csv", f"-{chunk_number}.csv")
+            df = pd.DataFrame()
+            df = pd.concat([df, chunk])
+            process_chunk(
+                df,
+                target_file_batch,
+                target_file,
+                chunk_number == 0,
+                chunk_number == 0,
+                output_headers,
+            )
+            logging.info(
+                f"Processing chunk #{chunk_number} of file {source_file} completed"
+            )
+
+
+def process_chunk(
+    df: pd.DataFrame,
+    target_file_batch: str,
+    target_file: str,
+    include_header: bool,
+    truncate_file: bool,
+    output_headers: typing.List[str],
+) -> None:
+    logging.info(f"Processing Batch {target_file_batch} started")
+    df = search_and_replace_values(df)
+    df = reorder_headers(df, output_headers)
+    save_to_new_file(df, file_path=str(target_file_batch))
+    append_file(target_file_batch, target_file, include_header, truncate_file, True)
+    logging.info(f"Processing Batch {target_file_batch} completed")
+
+
+def append_file(
+    batch_file_path: str,
+    target_file_path: str,
+    include_header: bool,
+    truncate_target_file: bool,
+    remove_source: bool = False,
+) -> None:
+    logging.info(
+        f"Appending file {batch_file_path} to file {target_file_path} with include_header={include_header} and truncate_target_file={truncate_target_file}"
+    )
+    data_file = open(batch_file_path, "r")
```

Comment:
It may be better to use these open(...) functions with "with" clauses.

Reply:
@happyhuman Done